File: 001201-001400/001400.构造K个回文字符串/1400.go
package main

import "fmt"

// 方法一：用时 8 ms
// 执行用时超过了 84 %
// 消耗内存超过了 60 %
// 难度: 中等
func canConstruct(s string, k int) bool {
	if len(s) < k {
		return false
	}
	cnts := make([]int, 26)
	for i := 0; i < len(s); i++ {
		cnts[s[i] - 'a'] ++
	}
	jnum := 0
	for i := 0; i < len(cnts); i++ {
		if cnts[i] % 2 == 1 {
			jnum ++
		}
	}
	return jnum <= k
}

func main ()  {
	fmt.Println(canConstruct("annabelle", 2))
	fmt.Println(canConstruct("leetcode", 3))
	fmt.Println(canConstruct("true", 4))
	fmt.Println(canConstruct("yzyzyzyzyzyzyzy", 2))
	fmt.Println(canConstruct("cr", 7))
}

File: 001201-001400/001400.构造K个回文字符串/1400.js
/**
 * @param {string} s
 * @param {number} k
 * @return {boolean}
 */
// 方法一：用时 100 ms
// 执行用时超过了 79 %
// 消耗内存超过了 67 %
// 难度: 中等
var canConstruct = function(s, k) {
  if (k > s.length) {
    return false
  }
  let cnts = {}
  for (let i = 0; i < s.length; i++) {
    if (cnts[s[i]]) {
      cnts[s[i]] ++
    } else {
      cnts[s[i]] = 1
    }
  }
  let jnum = 0
  for(let i in cnts) {
    if (cnts[i] % 2 != 0) {
      jnum ++
    }
  }
  return jnum <= k
};

console.log(canConstruct('annabelle', 2))
console.log(canConstruct('leetcode', 3))
console.log(canConstruct('true', 4))
console.log(canConstruct('yzyzyzyzyzyzyzy', 2))
console.log(canConstruct('cr', 7))
